test/test_python_dispatch.py

```diff
@@ -517,11 +514,83 @@ def test_enable_python_mode_respects_no_dispatch(self) -> None:
                 self.assertEqual(z.elem, expected)
 
     def test_nested_enable_python_mode(self) -> None:
-        with self.assertRaisesRegex(RuntimeError, "has already been set"):
+        class ErrorA(RuntimeError):
```

Comment:
All of this is not really needed since we just check that it fails to enable the mode? We could simply subclass LoggingTensor here.